### ISSUE TITLE
Unexpected Column production via hoist() and unnest_wider() on tibble()

### ISSUE THREAD

**Issue by npjc** [tidyverse/tidyr#782]
An empty `tibble()` behaves unexpectedly when subject to `under_wider()`. Specifically a new `..1` column is produced:

``` r
library(tibble)
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(tidyr)
```

This makes sense:

``` r
tibble(
    x = c("ok", "empty"), 
    y = list(
        tibble(ix = 1, iy = 2), 
        tibble()
    )
) %>% 
    unnest_wider(y)
#> # A tibble: 2 x 3
#>   x        ix    iy
#>   <chr> <dbl> <dbl>
#> 1 ok        1     2
#> 2 empty    NA    NA
```

but where does the extra `..1` col come from in the below?

``` r
tibble(
    x = c("ok", "empty"),
    y = list(
            list(a = tibble(ix = 1, iy = 2), b = tibble(iz = 3)),
            list(a = tibble(), b = tibble())
            )
    ) %>%
    hoist(y, a = "a") %>%
    select(-y) %>%
    unnest_wider(a)
#> New names:
#> * `` -> ...1
#> # A tibble: 2 x 4
#>   x        ix    iy ...1 
#>   <chr> <dbl> <dbl> <lgl>
#> 1 ok        1     2 NA   
#> 2 empty    NA    NA NA
```

<sup>Created on 2019-10-17 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by batpigandme** [comment]
Possibly related https://github.com/tidyverse/tidyr/issues/731#issuecomment-530953283

**Comment by hadley** [comment]
Can you please provide a minimal reproducible example using the [reprex](http://reprex.tidyverse.org) package? The goal of a reprex is to make it as easy as possible for me to recreate your problem so that I can fix it. If you've never made a minimal reprex before, there is lots of good advice [here](https://reprex.tidyverse.org/articles/reprex-dos-and-donts.html).

**Comment by hadley** [comment]
I guess you have used the reprex package, but you seem to have broken it up so I can't easily copy and paste into my R session. Please fix.

**Comment by hadley** [comment]
This now gives me:

``` r
library(tidyr)
tibble(
  x = c("ok", "empty"),
  y = list(
    list(a = tibble(ix = 1, iy = 2), b = tibble(iz = 3)),
    list(a = tibble(), b = tibble())
  )
) %>%
  hoist(y, a = "a") %>%
  dplyr::select(-y) %>%
  unnest_wider(a)
#> # A tibble: 2 x 3
#>   x        ix    iy
#>   <chr> <dbl> <dbl>
#> 1 ok        1     2
#> 2 empty    NA    NA
```

<sup>Created on 2019-11-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Was probably fixed by 7a72898c
